File: 25/11/go/1437.go
package C

func kLengthApart(nums []int, k int) bool {
	last := -k - 1
	for i := 0; i < len(nums); i++ {
		if nums[i] != 1 {
			continue
		}
		if i-last <= k {
			return false
		}
		last = i
	}
	return true
}

File: 25/11/1437.java
class Solution {
    public boolean kLengthApart(int[] nums, int k) {
        int last = -k - 1;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != 1) {
                continue;
            }
            if (i - last <= k) {
                return false;
            }
            last = i;
        }
        return true;
    }
}